contracts/MyStrategy.sol

```diff
@@ -453,18 +497,39 @@ contract MyStrategy is BaseStrategy, ReentrancyGuardUpgradeable {
         emit RewardsCollected(token, amount);
     }
 
-    /// @dev Send the BADGER token to the badgerTree
-    function _sendBadgerToTree(uint256 amount) internal {
-        // Transfer to tree without taking any fee
-        IERC20Upgradeable(BADGER).safeTransfer(IVault(vault).badgerTree(), amount);
-        emit TreeDistribution(BADGER, amount, block.number, block.timestamp);
+    /// @dev Takes a fee on the token and sends remaining to the given briber recepient
+    function _sendTokenToBriber(address token, address recepient, uint256 amount) internal {
+        // Process redirection fee
+        uint256 redirectionFee = amount.mul(redirectionFees[recepient]).div(MAX_BPS);
+        if (redirectionFee > 0) {
+            IERC20Upgradeable(token).safeTransfer(IVault(vault).treasury(), redirectionFee);
+            emit RedirectionFee(
+                IVault(vault).treasury(),
+                token,
+                redirectionFee,
+                block.number,
+                block.timestamp
+            );
+        }
+
+        // Send remaining to bribe recepient
+        // NOTE: Calculating instead of checking balance since there could have been an 
+        // existing balance on the contract beforehand
+        IERC20Upgradeable(token).safeTransfer(recepient, amount.sub(redirectionFee));
+        emit TokenRedirection(
+            recepient,
+            token,
+            amount.sub(redirectionFee),
+            block.number,
+            block.timestamp
+        );
     }
 
     function _sweepRewardToken(address token) internal {
         _onlyNotProtectedTokens(token);
 
         uint256 toSend = IERC20Upgradeable(token).balanceOf(address(this));
-        _handleRewardTransfer(token, toSend);
+        _sendTokenToBribesProcessor(token, toSend);
```

Comment:
Why this change? If someone else claims the bribes for us, then there would be no way to send them to the redirection contracts. We'd only be able to send it to the bribes processor

Reply:
Good call. I guess if we ever wanted to avoid redirection we would have to remove the route path beforehand through governance, which is a cleaner and more transparent process. Will revert the change, thanks!

Reply:
As a note, it just occurred to me that there may be a case in which a token is transferred to the strat by accident and this token has a redirection path set. In order to rescue this token we would have to atomically: remove path -> sweep -> add path back. Only issue with this is that it will be limited to governance only and not strategist but it is a very off scenario anyways.

Reply:
There's also a `sweepExtraToken` on the vault that might be useful for this